diffusion/evaluation/clean_fid_eval.py

```diff
@@ -0,0 +1,237 @@
+
+        # Init loggers
+        if self.loggers and dist.get_local_rank() == 0:
+            for logger in self.loggers:
+                if isinstance(logger, WandBLogger):
+                    wandb.init(**logger._init_kwargs)
+
+        # Load the model
+        Trainer(model=self.model,
+                load_path=self.load_path,
+                load_weights_only=True,
+                eval_dataloader=self.eval_dataloader)
+
+        # Move CLIP metric to device
+        self.device = dist.get_local_rank()
+        self.clip_metric = self.clip_metric.to(self.device)
+
+        # Predownload the CLIP model for computing clip-fid
+        _, _ = clip.load('ViT-B/32', device=self.device)
+
+    def _generate_images(self, guidance_scale: float):
+        """Core image generation function. Generates images at a given guidance scale.
+
+        Args:
+            guidance_scale (float): The guidance scale to use for image generation.
+        """
+        # Verify output dirs exist, if they don't, create them
+        real_image_path = os.path.join(self.output_dir, f'real_images_gs_{guidance_scale}')
+        gen_image_path = os.path.join(self.output_dir, f'gen_images_gs_{guidance_scale}')
+        if not os.path.exists(real_image_path) and dist.get_local_rank() == 0:
+            os.makedirs(real_image_path)
+        if not os.path.exists(gen_image_path) and dist.get_local_rank() == 0:
+            os.makedirs(gen_image_path)
+
+        # Reset the CLIP metric
+        self.clip_metric.reset()
+
+        # Storage for prompts
+        prompts = {}
+        # Iterate over the eval dataloader
+        num_batches = len(self.eval_dataloader)
+        for batch_id, batch in tqdm(enumerate(self.eval_dataloader)):
+            # Break if enough samples have been generated
+            if batch_id * self.batch_size * dist.get_world_size() >= self.num_samples:
+                break
+
+            real_images = batch['image']
+            captions = batch['captions']
+            # Ensure a new seed for each batch, as randomness in model.generate is fixed.
+            starting_seed = self.seed + num_batches * dist.get_local_rank()
+            seed = starting_seed + batch_id
+            # Generate images from the captions
+            with get_precision_context(self.precision):
+                generated_images = self.model.generate(tokenized_prompts=captions,
+                                                       height=self.size,
+                                                       width=self.size,
+                                                       guidance_scale=guidance_scale,
+                                                       seed=seed,
+                                                       progress_bar=False)  # type: ignore
+            # Get the prompts from the tokens
+            text_captions = [self.tokenizer.decode(caption, skip_special_tokens=True) for caption in captions]
```

Comment:
Nit: This should work, right?
```suggestion
            text_captions = self.tokenizer.batch_decode(captions, skip_special_tokens=True)
```

Reply:
Yes, it works too. Changed to this.